internal/testrunner/runners/system/runner.go

```diff
@@ -130,10 +142,140 @@ func (r *runner) TestFolderRequired() bool {
 	return true
 }
 
+// CanRunSetupTeardownIndependent returns whether this test runner can run setup or
+// teardown process independent.
+func (r *runner) CanRunSetupTeardownIndependent() bool {
+	return true
+}
+
+func createSetupServicesDir(elasticPackagePath *locations.LocationManager) error {
+	dirPath := elasticPackagePath.ServiceSetupDir()
+	err := os.MkdirAll(dirPath, 0755)
+	if err != nil {
+		return fmt.Errorf("mkdir failed (path: %s): %w", dirPath, err)
+	}
+	return nil
+}
+
 // Run runs the system tests defined under the given folder
 func (r *runner) Run(options testrunner.TestOptions) ([]testrunner.TestResult, error) {
 	r.options = options
-	return r.run()
+
+	if !r.options.RunSetup && !r.options.RunTearDown && !r.options.RunTestsOnly {
+		return r.run()
+	}
+
+	result := r.newResult("(init)")
+	if err := r.initRun(); err != nil {
+		return result.WithError(err)
+	}
+
+	if len(r.variants) > 1 {
+		return result.WithError(fmt.Errorf("a variant must be selected or trigger the test in no-variant mode (available variants: %s)", strings.Join(r.variants, ", ")))
+	}
+	if len(r.variants) == 0 {
+		logger.Debug("No variant mode")
+	}
+
+	_, err := os.Stat(r.locationManager.ServiceSetupDir())
+	if r.options.RunSetup && !os.IsNotExist(err) {
+		return result.WithError(fmt.Errorf("failed to run --setup, required to tear down previous setup run: %s exists", r.locationManager.ServiceSetupDir()))
+	}
+	if r.options.RunTestsOnly && os.IsNotExist(err) {
+		return result.WithError(fmt.Errorf("failed to run --no-provision, missing service setup folder: %s does not exist", r.locationManager.ServiceSetupDir()))
+	}
+	if r.options.RunTearDown && os.IsNotExist(err) {
+		return result.WithError(fmt.Errorf("failed to run --tear-down, missing service setup folder: %s does not exist", r.locationManager.ServiceSetupDir()))
+	}
```

Comment:
Maybe we can provide "higher-level" error messages, the user doesn't really matter about the existence of this file. We can keep the info about the service dir at the debug level.
```suggestion
	_, err := os.Stat(r.locationManager.ServiceSetupDir())
	logger.Debugf("Services setup data exists in %s: %v", r.locationManager.ServiceSetupDir(), !os.IsNotExist(err))
	if r.options.RunSetup && !os.IsNotExist(err) {
		return result.WithError(fmt.Errorf("failed to run --setup, required to tear down previous setup"))
	}
	if r.options.RunTestsOnly && os.IsNotExist(err) {
		return result.WithError(fmt.Errorf("failed to run tests with --no-provision, setup first with --setup"))
	}
	if r.options.RunTearDown && os.IsNotExist(err) {
		return result.WithError(fmt.Errorf("failed to run --tear-down, setup not found"))
	}
```